### PR TITLE
Add feature: user can add books to their shelf

### PR DESCRIPTION
**Pull request by ajtran303**
# Description

### New Migration

There is now a Book model instead of a Book poro.

### Feature

Users can add books to their shelves by going to the My Books page, and clicking the button.

# Closes issue(s)

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [x] I have tested this code
- [ ] I have updated the Readme

# Other comments
The flash messages are giving me a really hard time -- this may need some more work.
Additionally, these redirect paths may change (for example, should a user see a list of book results and then be able to pick a book that matches theirs?)

### CODE REVIEW

File: app/services/book_buddy_service.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class BookBuddyService
+  def self.find_by_isbn(isbn)
+    response = conn.get('/search') do |req|
+      req.params = {isbn: isbn}
+    end
+
+    if response.success?
+      JSON.parse(response.body, symbolize_names: true)
+    else
+      failure = { data: {
```
Comment (ajtran303) [inline app/services/book_buddy_service.rb:10]: We will want to just put this into the Sinatra side of things, eventually.

File: app/controllers/user/books_controller.rb [hunk 2]
```diff
@@ -4,4 +4,25 @@ class User::BooksController < ApplicationController
...
+  def create
+    # if current_user.nil? bounce to 404 page
+    
+    book = BookFacade.find_by_isbn(params[:isbn])
+    if book.isbn.nil?
+      flash[:failure] = "Oops! That book cannot be found."
+      redirect_to new_user_book_path and return
+    end
+
+    user_book = UserBook.new(status: 'available', user_id: current_user.id, isbn: book.isbn)
```
Comment (ljwhite) [inline app/controllers/user/books_controller.rb:19]: Future work: have the user confirm to add this book to user shelf. Right now it just adds the book, even if user accidentally inserted wrong ISBN

File: spec/features/books/user_books_index_spec.rb [hunk 3]
```diff
@@ -5,175 +5,8 @@
...
-          "searchInfo": {
-            "textSnippet": 'An expert at simulated war games, Andrew &quot;Ender&quot; Wiggin believes that he is engaged in one more computer war game when, in truth, he is commanding the last Earth fleet against an alien race seeking Earth&#39;s complete destruction.'
-          }
-        }
-      ]
-    }
-    json_call2 = api_call2.to_json
-    book_attrs2 = JSON.parse(json_call2, symbolize_names: true)
-    @book2 = Book.new(book_attrs2)
+    @book1 = create(:book)
```
Comment (ljwhite) [inline spec/features/books/user_books_index_spec.rb:8]: so much easier to test for books!

**Comment by ljwhite** [comment]
Looks good. We will need to have the user confirm that the search result pulled up the expected book before user adds it to the shelf.

**Pull request merged**